app/src/androidTest/java/saveServiceTests/StartStopService.java

```diff
@@ -0,0 +1,133 @@
+                "Successfully started 'Save Location Service'  Via Init Module");
+
+        boolean isLocationPingServiceRunning = UiUtils.isServiceRunning(LocationPingService.class.getName());
+
+        AssertUtils.assertTrueV(isLocationPingServiceRunning,
+                "Failed to start 'Ping Location Service' Via Init Module",
+                "Successfully started 'Ping Location Service'  Via Init Module");
+
+
+        // ----------------------------------------------------------------------------------------------------------------------------------
+
+
+        LogUITest.debug("Stopping 'Save Location Service' & 'Ping Location Service Via Init Module'");
+        LocationsHelper.INSTANCE.stopAndClearAll(activityTestRule.getActivity().getApplication());
+
+        isLocationSaveServiceRunning = UiUtils.isServiceRunning(LocationSaveService.class.getName());
+
+        AssertUtils.assertTrueV(!isLocationSaveServiceRunning,
+                "Failed to stop 'Save Location Service'  Via Init Module",
+                "Successfully stopped 'Save Location Service'  Via Init Module");
+
+        isLocationPingServiceRunning = UiUtils.isServiceRunning(LocationPingService.class.getName());
+
+        AssertUtils.assertTrueV(!isLocationPingServiceRunning,
+                "Failed to stop 'Ping Location Service' Via Init Module",
+                "Successfully stopped 'Ping Location Service'  Via Init Module");
+
+
+    }
+
+
+    @AutoTest_StartStopServices
+    @Test
+    public void verifyStartStopLocationPingService() {
+
+        // Start Location Ping Service
+        LogUITest.debug("Starting 'Ping Location Service'");
+        LocationsHelper.INSTANCE.startLocationPingService(activityTestRule.getActivity().getApplication(), null, locationConfigs);
+
+        boolean isLocationSaveServiceRunning = UiUtils.isServiceRunning(LocationPingService.class.getName());
+
+        AssertUtils.assertTrueV(isLocationSaveServiceRunning,
+                "Failed to start 'Ping Location Service'",
+                "Successfully started 'Ping Location Service'");
+
+
+        // Stop Location Ping Service
+        LogUITest.debug("Stopping 'Ping Location Service'");
+        LocationsHelper.INSTANCE.stopLocationPingService(activityTestRule.getActivity().getApplication());
+
+        boolean isServiceRunning = UiUtils.isServiceRunning(LocationPingService.class.getName());
+
+        AssertUtils.assertTrueV(!isServiceRunning,
+                "Failed to stop 'Ping Location Service'",
+                "Successfully stopped 'Ping Location Service'");
+
+
+    }
+
+
+    @AutoTest_StartStopServices
+    @Test
+    public void verifyStartStopLocationSaveService() {
+
+        // Start Location Save Service
+        LogUITest.debug("Starting 'Save Location Service'");
+        LocationsHelper.INSTANCE.startLocationSaveService(activityTestRule.getActivity().getApplication(), locationConfigs);
```

Comment:
Why do we not use the init() method which the apps use to start GPS SDK for all these start-stop tests?

Reply:
So there are 3 different tests . 
![image](https://user-images.githubusercontent.com/38068550/77505255-ac16bb80-6e88-11ea-9a54-d95c5a2a452a.png)

Kindly read the description of very first test. 
First test is all about starting services from init module (which other apps call to start the services)